include/bit_algorithm_details.hpp

```diff
@@ -92,6 +96,138 @@ T get_word(bit_iterator<InputIt> first, T len=binary_digits<T>::value)
     return ret_word;
 }
 
+// Get next len bits beginning at start and store them in a word of type T
+// If we reach `last` before we get len bits, break and return the current word
+// bits_read will store the number of bits that we read.
+template <class T, class InputIt>
+T get_word(bit_iterator<InputIt> first, bit_iterator<InputIt> last,
+        T& bits_read, T len=binary_digits<T>::value
+        )
+{
+    using native_word_type = typename bit_iterator<InputIt>::word_type;
+    constexpr T native_digits = binary_digits<native_word_type>::value; 
+    constexpr T ret_digits = binary_digits<T>::value; 
+    assert(ret_digits >= len);
+    bits_read = native_digits - first.position();
+    T ret_word = *first.base() >> first.position();
```

Comment:
i'm scared of this now. if `*first.base()` is a short there might be weird behavior?

Reply:
So I think here it is immediately casted to to `T`, so there is no issue. I think its when you use the results in intermediate operations where issues arise.